### ISSUE TITLE
JSON client class compilation failure

### ISSUE THREAD

**Issue by stoicflame** [stoicflame/enunciate#255]
hey there.  I am trying to use enunciate 2.2.0 with a maven project, and I'm getting the following:

```
Failed to execute goal com.webcohesion.enunciate:enunciate-maven-plugin:2.2.0:assemble (assemble) on project jjj-rest-war: Execution assemble of goal com.webcohesion.enunciate:enunciate-maven-plugin:2.2.0:assemble failed: com.webcohesion.enunciate.EnunciateException: Compile failed of Java JSON client-side classes. OnError while emitting onNext value: com.webcohesion.enunciate.EnunciateContext.class -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal com.webcohesion.enunciate:enunciate-maven-plugin:2.2.0:assemble (assemble) on project jjj-rest-war: Execution assemble of goal com.webcohesion.enunciate:enunciate-maven-plugin:2.2.0:assemble failed: com.webcohesion.enunciate.EnunciateException: Compile failed of Java JSON client-side classes.
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:224)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:108)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:76)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:116)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:361)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:155)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:584)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:213)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:157)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: org.apache.maven.plugin.PluginExecutionException: Execution assemble of goal com.webcohesion.enunciate:enunciate-maven-plugin:2.2.0:assemble failed: com.webcohesion.enunciate.EnunciateException: Compile failed of Java JSON client-side classes.
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:144)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
	... 19 more
Caused by: java.lang.RuntimeException: com.webcohesion.enunciate.EnunciateException: Compile failed of Java JSON client-side classes.
	at com.sun.tools.javac.main.Main.compile(Main.java:553)
	at com.sun.tools.javac.api.JavacTaskImpl.doCall(JavacTaskImpl.java:129)
	at com.sun.tools.javac.api.JavacTaskImpl.call(JavacTaskImpl.java:138)
	at com.webcohesion.enunciate.Enunciate.run(Enunciate.java:656)
	at com.webcohesion.enunciate.mojo.ConfigMojo.execute(ConfigMojo.java:368)
	at com.webcohesion.enunciate.mojo.AssembleBaseMojo.execute(AssembleBaseMojo.java:86)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:133)
	... 20 more
Caused by: com.webcohesion.enunciate.EnunciateException: Compile failed of Java JSON client-side classes.
	at com.webcohesion.enunciate.modules.java_json_client.JavaJSONClientModule.compileClientSources(JavaJSONClientModule.java:317)
	at com.webcohesion.enunciate.modules.java_json_client.JavaJSONClientModule.call(JavaJSONClientModule.java:121)
	at com.webcohesion.enunciate.io.InvokeEnunciateModule.onNext(InvokeEnunciateModule.java:31)
	at com.webcohesion.enunciate.io.InvokeEnunciateModule.onNext(InvokeEnunciateModule.java:10)
	at rx.internal.operators.OperatorDoOnEach$1.onNext(OperatorDoOnEach.java:79)
	at rx.internal.operators.OperatorSingle$ParentSubscriber.onCompleted(OperatorSingle.java:109)
	at rx.internal.operators.NotificationLite.accept(NotificationLite.java:140)
	at rx.internal.operators.TakeLastQueueProducer.emit(TakeLastQueueProducer.java:100)
	at rx.internal.operators.TakeLastQueueProducer.startEmitting(TakeLastQueueProducer.java:45)
	at rx.internal.operators.OperatorTakeLast$1.onCompleted(OperatorTakeLast.java:59)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.drainAndComplete(OperatorMerge.java:521)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.completeInner(OperatorMerge.java:486)
	at rx.internal.operators.OperatorMerge$InnerSubscriber.emit(OperatorMerge.java:658)
	at rx.internal.operators.OperatorMerge$InnerSubscriber.onCompleted(OperatorMerge.java:600)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.drainAndComplete(OperatorMerge.java:521)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.completeInner(OperatorMerge.java:486)
	at rx.internal.operators.OperatorMerge$InnerSubscriber.emit(OperatorMerge.java:673)
	at rx.internal.operators.OperatorMerge$InnerSubscriber.onCompleted(OperatorMerge.java:600)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.drainAndComplete(OperatorMerge.java:521)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.completeInner(OperatorMerge.java:486)
	at rx.internal.operators.OperatorMerge$InnerSubscriber.emit(OperatorMerge.java:673)
	at rx.internal.operators.OperatorMerge$InnerSubscriber.onCompleted(OperatorMerge.java:600)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.drainAndComplete(OperatorMerge.java:521)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.completeInner(OperatorMerge.java:486)
	at rx.internal.operators.OperatorMerge$InnerSubscriber.emit(OperatorMerge.java:673)
	at rx.internal.operators.OperatorMerge$InnerSubscriber.onCompleted(OperatorMerge.java:600)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.drainAndComplete(OperatorMerge.java:521)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.completeInner(OperatorMerge.java:486)
	at rx.internal.operators.OperatorMerge$InnerSubscriber.emit(OperatorMerge.java:673)
	at rx.internal.operators.OperatorMerge$InnerSubscriber.onCompleted(OperatorMerge.java:600)
	at rx.internal.util.ScalarSynchronousObservable$ScalarSynchronousAction.call(ScalarSynchronousObservable.java:117)
	at rx.schedulers.ExecutorScheduler$ExecutorAction.run(ExecutorScheduler.java:173)
	at rx.schedulers.ExecutorScheduler$ExecutorSchedulerWorker.run(ExecutorScheduler.java:99)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: rx.exceptions.OnErrorThrowable$OnNextValue: OnError while emitting onNext value: com.webcohesion.enunciate.EnunciateContext.class
	at rx.exceptions.OnErrorThrowable.addValueAsLastCause(OnErrorThrowable.java:101)
	at rx.internal.operators.OperatorDoOnEach$1.onNext(OperatorDoOnEach.java:81)
	... 31 more


For more information about the errors and possible solutions, please read the following articles:
[Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginExecutionException
[ENUNCIATE] /Users/darkos/yyy-api/yyy-api-project/trunk/modules/yyy-rest-common/war/target/enunciate/java-xml-client/src is NOT up-to-date because it's an empty directory.
[ENUNCIATE] Generating the Java client classes...
[ENUNCIATE] Processing template null.
Exception in thread "pool-1-thread-1" java.lang.NoClassDefFoundError: com/webcohesion/enunciate/modules/java_xml_client/JavaXMLClientModule$2
	at com.webcohesion.enunciate.modules.java_xml_client.JavaXMLClientModule.processTemplate(JavaXMLClientModule.java:305)
	at com.webcohesion.enunciate.modules.java_xml_client.JavaXMLClientModule.generateClientSources(JavaXMLClientModule.java:253)
	at com.webcohesion.enunciate.modules.java_xml_client.JavaXMLClientModule.call(JavaXMLClientModule.java:123)
	at com.webcohesion.enunciate.io.InvokeEnunciateModule.onNext(InvokeEnunciateModule.java:31)
	at com.webcohesion.enunciate.io.InvokeEnunciateModule.onNext(InvokeEnunciateModule.java:10)
	at rx.internal.operators.OperatorDoOnEach$1.onNext(OperatorDoOnEach.java:79)
	at rx.internal.operators.OperatorSingle$ParentSubscriber.onCompleted(OperatorSingle.java:109)
	at rx.internal.operators.NotificationLite.accept(NotificationLite.java:140)
	at rx.internal.operators.TakeLastQueueProducer.emit(TakeLastQueueProducer.java:100)
	at rx.internal.operators.TakeLastQueueProducer.startEmitting(TakeLastQueueProducer.java:45)
	at rx.internal.operators.OperatorTakeLast$1.onCompleted(OperatorTakeLast.java:59)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.drainAndComplete(OperatorMerge.java:521)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.completeInner(OperatorMerge.java:486)
	at rx.internal.operators.OperatorMerge$InnerSubscriber.emit(OperatorMerge.java:658)
	at rx.internal.operators.OperatorMerge$InnerSubscriber.onCompleted(OperatorMerge.java:600)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.drainAndComplete(OperatorMerge.java:521)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.completeInner(OperatorMerge.java:486)
	at rx.internal.operators.OperatorMerge$InnerSubscriber.emit(OperatorMerge.java:673)
	at rx.internal.operators.OperatorMerge$InnerSubscriber.onCompleted(OperatorMerge.java:600)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.drainAndComplete(OperatorMerge.java:521)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.completeInner(OperatorMerge.java:486)
	at rx.internal.operators.OperatorMerge$InnerSubscriber.emit(OperatorMerge.java:673)
	at rx.internal.operators.OperatorMerge$InnerSubscriber.onCompleted(OperatorMerge.java:600)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.drainAndComplete(OperatorMerge.java:521)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.completeInner(OperatorMerge.java:486)
	at rx.internal.operators.OperatorMerge$InnerSubscriber.emit(OperatorMerge.java:673)
	at rx.internal.operators.OperatorMerge$InnerSubscriber.onCompleted(OperatorMerge.java:600)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.drainAndComplete(OperatorMerge.java:521)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.completeInner(OperatorMerge.java:486)
	at rx.internal.operators.OperatorMerge$InnerSubscriber.emit(OperatorMerge.java:673)
	at rx.internal.operators.OperatorMerge$InnerSubscriber.onCompleted(OperatorMerge.java:600)
	at rx.internal.util.ScalarSynchronousObservable$ScalarSynchronousAction.call(ScalarSynchronousObservable.java:117)
	at rx.schedulers.ExecutorScheduler$ExecutorAction.run(ExecutorScheduler.java:173)
	at rx.schedulers.ExecutorScheduler$ExecutorSchedulerWorker.run(ExecutorScheduler.java:99)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

here is what I have in my pom, for the plugin: enunciate.version is 2.2.0 by the way....

```xml
            <plugin>
                <groupId>com.webcohesion.enunciate</groupId>
                <artifactId>enunciate-maven-plugin</artifactId>
                <version>${enunciate.version}</version>
                <executions>
                    <execution>
                        <id>assemble</id>
                        <goals>
                            <goal>assemble</goal>
                        </goals>
                        <configuration>
                            <project-extensions>
                                <extension>java-json-client</extension>
                                <extension>java-xml-client</extension>
                            </project-extensions>
                        </configuration>
                    </execution>
                </executions>
            </plugin>
```

I am using jdk 8 u 66.

(Originally reported at #138 by @ke5stl.)

**Comment by stoicflame** [comment]
@ke5stl , we're moving your issue to a new thread here. This is different from the `NoClassDefFoundError`s at #138, which is being used to track `NoClassDefFoundError`s that have no other output.

This particular case is a compile failure of your client-side JSON classes. Unfortunately, you didn't include the compiler errors, which should have been reported earlier in your build. Will you please include those here?

**Comment by ke5stl** [comment]
ah....

here we go:

warning: [options] To suppress warnings about obsolete options, use -Xlint:-options.
/Users/pbelbin/nbp-api/bcpbx-api-project/trunk/modules/bcpbx-rest-common/war/target/enunciate/java-xml-client/src/com/broadsoft/cpbx/api/rialto/domain/RequestedFOCDate.java:21: error: cannot inherit from final RialtoDate
public class RequestedFOCDate extends com.broadsoft.cpbx.api.rialto.domain.RialtoDate {

**Comment by stoicflame** [comment]
What does the `RequestedFOCDate` look like?

**Comment by ke5stl** [comment]
It is a basic pojo with get/setters i think, but it does make use of a
utility class static method to ingest/validate/standardize the set value, i
think.   I am mobile currently.

**Comment by ke5stl** [comment]
Oh, i think it extends a base type, or implements an interface definition
object.

**Comment by ke5stl** [comment]
ok.  here is what I think you need:

package com.broadsoft.cpbx.api.rialto.domain;

/**
 * 
 @author pbelbin

 */
@javax.xml.bind.annotation.XmlType (
  name = "requestedFOCDate",
  namespace = ""
)
@javax.xml.bind.annotation.XmlRootElement (
  name = "requestedFOCDate",
  namespace = ""
)
public class RequestedFOCDate extends com.broadsoft.cpbx.api.rialto.domain.RialtoDate {


}


and here is RialtoDate:

@javax.xml.bind.annotation.XmlType (
  name = "rialtoDate",
  namespace = ""
)
@javax.xml.bind.annotation.XmlRootElement (
  name = "rialtoDate",
  namespace = ""
)
public final class RialtoDate implements java.io.Serializable {

  private java.lang.String _value;

  public RialtoDate() {
  }

  /**
   * (no documentation provided)
   */
  @javax.xml.bind.annotation.XmlValue
  public java.lang.String getValue() {
    return this._value;
  }

  /**
   * (no documentation provided)
   */
  public void setValue(java.lang.String _value) {
    this._value = _value;
  }

}

**Comment by ke5stl** [comment]
by the way, I changed the source object RialtoDate to not be final, but, I see that the generated source under the war project target/enunciate source object has the final, and it's still failing.

**Comment by ke5stl** [comment]
interestingly, the enunciate version of this object (RialtoDate) under the java-json-client tree is not final.

**Comment by ke5stl** [comment]
Ok.  I've created a sample based on the IfYouWannaBeCool sample app.

I've attached the project source.

Note, I've found 2 issues.

1.  in the CleanDate object, the @XmlValue annotation on the date attribute appears to be causing the issue that I've been seeing.
2. on the way to creating this sample, I discovered another issue, which is that it seems that there's a problem if there is not a setter that takes the same input type as the constructor, particularly when there are overloaded constructors.

try doing a maven clean install to see the issue that started this thread.

modify CleanDate to see the 2nd issue.

please advise.

[enunciate-sample-master.tar.gz](https://github.com/stoicflame/enunciate/files/95147/enunciate-sample-master.tar.gz)

thank you!

**Comment by ke5stl** [comment]
hello there!  were you able to reproduce the issue I'm seeing?  I appreciate the help.

**Comment by stoicflame** [comment]
Confirmed.

So what's the purpose of the `DateHired` class? Usually, you extend classes so that you can add properties, but the JAXB spec says that you can't add properties to a class that has an `@XmlValue`, so Enunciate's generated client-side classes make those final.

**Comment by ke5stl** [comment]
the DateHired class is present to allow the restful payload to use a more descriptive name for the value.
there may also be cases where multiple subtypes of the same base type could be in the same payload, each named distinctly.  eg: DateOfBirth, DateHired, might both need to be in the same payload, and they're all the same underlying type.

**Comment by stoicflame** [comment]
Okay, I'll fix it for 2.3.

**Comment by ke5stl** [comment]
much appreciated!

**Comment by ke5stl** [comment]
mmmmm.  what about the issue that is manifest when there are overloads on the constructor?

**Comment by stoicflame** [comment]
I'll look at that, too.

**Comment by ke5stl** [comment]
I tried modifying the java-xml-client client-simple-type.fmt line 25 by removing the 'final' qualifier, and this seems to allow enunciate to complete, as long as I also remove the overloaded constructor and accessor methods.

**Comment by ke5stl** [comment]
might also be an issue with overloaded getters/setters.

**Comment by ke5stl** [comment]
@XmlRootElement
public class MyBoolean {

    @XmlValue
    @NotNull
    private Boolean value;

    /**
     * @return the value
     */
    public Boolean getValue() {
        return value;
    }

    /**
     * @param value the value to set
     */
    public void setValue(Boolean value) {
        this.value = value;
    }

}

appears to have problems, resulting in:

error: variable _value is already defined in class
error: method getValue() is already defined in class
error: method setValue(Boolean) is already defined in class 

seems that the generation of the class is doubling up

**Comment by ke5stl** [comment]
it appears that the issues created by use of @Xml{whatever} is more widespread.

eg: if I annotate an element as being @XmlElement(nillable=true) then the build fails, complaining that the variable has already been defined.

**Comment by stoicflame** [comment]
I fixed the "final" compile failures at 7f0d4b5.

However, I need more direction on the other problem. Enunciate could be more descriptive in it's error processing, but the root of the problem is that you've got conflicting members. You've got one member named "date" which is the _field_ annotated with `@XmlValue` (or `@Xml{whatever}`). You've got _another_ member named "date" which is the _property_ (getter/setter pair).

If you've decided you want to use field-based accessors and NOT property-based accessors, you need to annotate your class or package with `@XmlAccessorType(XmlAccessType.FIELD)` or `@XmlAccessorType(XmlAccessType.NONE)` so that the properties don't get picked up as members.

If I try to use JAXB to write out the `CleanDate` class, I get this error:

```
com.sun.xml.internal.bind.v2.runtime.IllegalAnnotationsException: 2 counts of IllegalAnnotationExceptions

If a class has @XmlElement property, it cannot have @XmlValue property.
	this problem is related to the following location:
		at private java.lang.String org.codehaus.enunciate.jaxb.ri.CleanDate.date
		at org.codehaus.enunciate.jaxb.ri.CleanDate
	this problem is related to the following location:
		at public java.lang.String org.codehaus.enunciate.jaxb.ri.CleanDate.getDate()
		at org.codehaus.enunciate.jaxb.ri.CleanDate
Class has two properties of the same name "date"
	this problem is related to the following location:
		at public java.lang.String org.codehaus.enunciate.jaxb.ri.CleanDate.getDate()
		at org.codehaus.enunciate.jaxb.ri.CleanDate
	this problem is related to the following location:
		at private java.lang.String org.codehaus.enunciate.jaxb.ri.CleanDate.date
		at org.codehaus.enunciate.jaxb.ri.CleanDate

	at com.sun.xml.internal.bind.v2.runtime.IllegalAnnotationsException$Builder.check(IllegalAnnotationsException.java:91)
	at com.sun.xml.internal.bind.v2.runtime.JAXBContextImpl.getTypeInfoSet(JAXBContextImpl.java:436)
	at com.sun.xml.internal.bind.v2.runtime.JAXBContextImpl.<init>(JAXBContextImpl.java:277)
	at com.sun.xml.internal.bind.v2.runtime.JAXBContextImpl$JAXBContextBuilder.build(JAXBContextImpl.java:1100)
	at com.sun.xml.internal.bind.v2.ContextFactory.createContext(ContextFactory.java:143)
	at com.sun.xml.internal.bind.v2.ContextFactory.createContext(ContextFactory.java:110)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
	at java.lang.reflect.Method.invoke(Method.java:597)
	at javax.xml.bind.ContextFinder.newInstance(ContextFinder.java:202)
	at javax.xml.bind.ContextFinder.find(ContextFinder.java:376)
	at javax.xml.bind.JAXBContext.newInstance(JAXBContext.java:574)
	at javax.xml.bind.JAXBContext.newInstance(JAXBContext.java:522)
```

So for now, I'll just try to make the error easier to understand. Yes?

**Comment by ke5stl** [comment]
Let me take another look at the sample, and I’ll see if I can take what you’ve said into account.

 

The other issue was to do with overloaded constructors, or was it just overloaded getters/setters?

 

It seemed like there was an assumption that for a given getter/setter pair, that there should be a single attribute with the name of the getter/setter method.

 

However, with overloaded methods, where the name of the method is the same, another attribute is being generated, again, with the same name, and this is causing issues.

 

Seems like the attribute name needs to be a composite of the method name + param or return type, so that you can have multiple without colliding.  

 

I wonder, if the object you create were created entirely without the private members, and have the xml generation use the methods instead of attributes, then you wouldn’t even have to generate any private attributes at all, and avoid the issue entirely?

**Comment by ke5stl** [comment]
hmmmm.  or, perhaps only one of the overloaded methods should really be
considered as candidates for generation into the enunciate version of the
class in question?

after all, that's kinda what the xml marshaling code requires (eg: jaxb), I
think, at least in some cases.

?

thank you for working on this!

**Comment by stoicflame** [comment]
I know of no issue with overloaded constructors. Enunciate doesn't even consider constructors in its processing engine.

I know of no issue with overloaded methods; Enunciate won't generate duplicate attributes of the same name for overloaded setters.

The only issue I'm aware of is when you've got both a field _and_ a property of the same name. And it's not just Enunciate that has a problem with this; it's JAXB, too (as you can see from the `IllegalAnnotationsException` that manifests itself).

So let's take a class that looks like this, which has one _field_ named "date" and one _property_ named "date" defined by the getter/setter pair.

```java
@XmlRootElement
public class CleanDate {
  private String date;
  public String getDate() { return date; }
  public void setDate(String date) { this.date = date }
}
```

JAXB handles this class because, by default, JAXB considers all `PUBLIC_MEMBER`s as accessors. Since the only public member in this case is the property (the field is private), there's only one accessor named "date".

However, if I add an explicit JAXB annotation to the _field_:

```java
@XmlRootElement
public class CleanDate {
  @XmlValue
  private String date;
  public String getDate() { return date; }
  public void setDate(String date) { this.date = date }
}
```

Then JAXB picks up the _field_ because it's explicitly annotated and it picks up the _property_ because it's a `PUBLIC_MEMBER`. Then it sees that there's a naming conflict: both the _field_ it picked up and the _property_ it picked up are named "date" and it throws the `IllegalAnnotationsException`.

So I can suggest two ways to avoid the `IllegalAnnotationsException`. The first is to annotate the _property_ instead of the _field_:

```java
@XmlRootElement
public class CleanDate {
  private String date;
  @XmlValue
  public String getDate() { return date; }
  public void setDate(String date) { this.date = date }
}
```

That way, JAXB will continue to ignore the field.

The other suggestion is to tell JAXB to _not_ use it's default `PUBLIC_MEMBER` policy for picking up accessors, but to use a _different_ policy, like `FIELD` or `NONE`:

```java
@XmlAccessorType(XmlAccessType.FIELD) //or @XmlAccessorType(XmlAccessType.NONE)
@XmlRootElement
public class CleanDate {
  @XmlValue
  private String date;
  public String getDate() { return date; }
  public void setDate(String date) { this.date = date }
}
```

**Comment by stoicflame** [comment]
Better error handling has been implemented at ba1f06b.

**Comment by stoicflame** [comment]
[Enunciate 2.3 has been released](https://github.com/stoicflame/enunciate/releases/tag/v2.3.0).
